src/applications/ivc-champva/10-7959C/chapters/healthInsuranceInformation.js

```diff
@@ -77,32 +83,36 @@ export const applicantProviderSchema = isPrimary => {
       [keyname]: { type: 'string' },
     }),
   };
-};
+}
 
-export const applicantPrimaryEffectiveDateSchema = {
-  uiSchema: {
-    applicants: {
-      'ui:options': {
-        viewField: ApplicantField,
-      },
-      items: {
-        ...titleUI(
-          ({ formData }) =>
-            `${applicantWording(formData)} ${
-              formData?.applicantPrimaryProvider
-            } insurance effective date`,
-        ),
-        applicantPrimaryEffectiveDate: currentOrPastDateUI(
-          'Health insurance effective date',
-        ),
+export function applicantInsuranceEffectiveDateSchema(isPrimary) {
+  const keyname = `applicant${
+    isPrimary ? 'Primary' : 'Secondary'
+  }EffectiveDate`;
+  return {
+    uiSchema: {
+      applicants: {
+        'ui:options': {
+          viewField: ApplicantField,
+        },
+        items: {
+          ...titleUI(
+            ({ formData }) =>
+              `${applicantWording(formData)} ${insuranceName(
+                formData,
+                isPrimary,
+              )} insurance effective date`,
+          ),
+          [keyname]: currentOrPastDateUI('Health insurance effective date'),
+        },
       },
     },
-  },
-  schema: applicantListSchema(['applicantPrimaryEffectiveDate'], {
-    titleSchema,
-    applicantPrimaryEffectiveDate: currentOrPastDateSchema,
-  }),
-};
+    schema: applicantListSchema([keyname], {
+      titleSchema,
+      [keyname]: currentOrPastDateSchema,
+    }),
+  };
+}
```

Comment:
Reworked this schema to work for both primary and secondary insurance pages based on bool passed in.